BParkingNano/plugins/TrackMerger.cc

```diff
@@ -229,13 +222,24 @@ void TrackMerger::produce(edm::StreamID, edm::Event &evt, edm::EventSetup const
     if ( iTrk < nTracks )
       pcand.addUserCand( "cand", edm::Ptr<pat::PackedCandidate> ( tracks, iTrk ));
     else 
-      pcand.addUserCand( "cand", edm::Ptr<pat::PackedCandidate> ( lostTracks, iTrk-nTracks ));
-    
-    tracks_out       -> emplace_back(pcand);
-    trans_tracks_out -> emplace_back(trackTT);
-   
-  }
+      pcand.addUserCand( "cand", edm::Ptr<pat::PackedCandidate> ( lostTracks, iTrk-nTracks ));   
  
+  //in order to avoid revoking the sxpensive ttrack builder many times and still have everything sorted, we add them to vector of pairs
+   vectrk_ttrk.emplace_back( std::make_pair(pcand,trackTT ) );   
+  }
+
+  // sort to be uniform with leptons
+  std::sort( vectrk_ttrk.begin(), vectrk_ttrk.end(), 
+             [] ( auto & trk1, auto & trk2) -> 
+                  bool {return (trk1.first).pt() > (trk2.first).pt();} 
+           );
```

Comment:
to speed it up slightly, maybe worth changing vectrk_ttrk in a pair of <float, TTtrack> with (pt, tttrack). So we access the pt once per candidate when filling the vectrk_ttrk (L228) )and then have it ready for the sorting